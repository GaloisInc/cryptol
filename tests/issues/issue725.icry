:set show-examples = no
:set prover-stats=no

:set prover=z3

:prove \(n:Integer, p : Bit) -> p ==> (x != x where x = 2 ^^ n)
:sat \(a:[8], b:[8]) -> a == ~zero /\ a@b == False
:prove \(a:[8], b:[8]) -> a == zero ==> a@b == False

:set prover=w4-z3

:prove \(n:Integer, p : Bit) -> p ==> (x != x where x = 2 ^^ n)
:sat \(a:[8], b:[8]) -> a == ~zero /\ a@b == False
:prove \(a:[8], b:[8]) -> a == zero ==> a@b == False
