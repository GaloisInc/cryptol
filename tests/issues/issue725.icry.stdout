Loading module Cryptol

operation can not be supported on symbolic values: integer exponentiation
Unsatisfiable
Counterexample

operation can not be supported on symbolic values: integer exponentiation
Unsatisfiable
Counterexample
